core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -1555,30 +1632,47 @@ private String formatDomain(Domain domain)
             return "[" + Joiner.on(", ").join(parts.build()) + "]";
         }
 
+        private String formatFilter(Expression filter)
+        {
+            return filter.equals(TRUE_LITERAL) ? "" : filter.toString();
+        }
+
+        private String formatBoolean(boolean value)
+        {
+            return value ? "true" : "";
+        }
+
+        private <T> String formatCollection(Collection<T> collection)
+        {
+            return collection.stream()
+                    .map(Objects::toString)
+                    .collect(joining(", ", "[", "]"));
+        }
+
         public NodeRepresentation addNode(PlanNode node, String name)
         {
-            return addNode(node, name, "");
+            return addNode(node, name, ImmutableMap.of());
         }
 
-        public NodeRepresentation addNode(PlanNode node, String name, String identifier)
+        public NodeRepresentation addNode(PlanNode node, String name, Map<String, String> identifier)
         {
             return addNode(node, name, identifier, node.getSources(), Optional.empty());
         }
 
-        public NodeRepresentation addNode(PlanNode node, String name, String identifier, Optional<PlanNodeStatsAndCostSummary> reorderJoinStatsAndCost)
+        public NodeRepresentation addNode(PlanNode node, String name, Map<String, String> identifier, Optional<PlanNodeStatsAndCostSummary> reorderJoinStatsAndCost)
         {
             return addNode(node, name, identifier, node.getSources(), reorderJoinStatsAndCost);
         }
 
-        public NodeRepresentation addNode(PlanNode node, String name, String identifier, List<PlanNode> children, Optional<PlanNodeStatsAndCostSummary> reorderJoinStatsAndCost)
+        public NodeRepresentation addNode(PlanNode node, String name, Map<String, String> identifier, List<PlanNode> children, Optional<PlanNodeStatsAndCostSummary> reorderJoinStatsAndCost)
         {
             return addNode(node, name, identifier, ImmutableList.of(node.getId()), children, ImmutableList.of(), reorderJoinStatsAndCost);
         }
 
         public NodeRepresentation addNode(
                 PlanNode rootNode,
                 String name,
-                String identifier,
+                Map<String, String> identifier,
```

Comment:
`identifier` name seems misleading. Maybe rename it to `descriptor`.

Consider adding dedicated `Descriptor` type so that `Map<String, String>` is not pushed everywhere

Reply:
> Consider adding dedicated Descriptor type so that Map<String, String> is not pushed everywhere

What do you mean? Do you mean something like Map<String, Object>?

Reply:
I mean Descriptor object that contains map. See Symbol or PlanNodeId. They are just wrappers over raw type so that it's easy to distinguish what given parameter or field represents.

Consider that optional though.  However changing fields and params to "descriptor"  would be nice to have. Identifier seems like a bad name here (especially when it became map)